packages/blocks/src/root-block/edgeless/components/toolbar/edgeless-toolbar.ts

```diff
@@ -585,21 +308,28 @@ export class EdgelessToolbar extends WithDisposable(LitElement) {
       return nothing;
     }
 
-    const Content =
-      type === 'frameNavigator' ? this._FrameNavigator : this._DefaultContent;
-
     return html`
       <div
         class="edgeless-toolbar-toggle-control"
-        data-hide=${this._hideNavigatorToolbar}
+        data-enable=${this._enableAutoHide}
       >
         <div
           class="edgeless-toolbar-container"
           @dblclick=${stopPropagation}
           @mousedown=${stopPropagation}
           @pointerdown=${stopPropagation}
         >
-          ${Content}
+          <presentation-toolbar
+            .visible=${type === 'frameNavigator'}
+            .edgeless=${this.edgeless}
+            .settingMenuShow=${this.presentSettingMenuShow}
+            .frameMenuShow=${this.presentFrameMenuShow}
+            .setSettingMenuShow=${(show: boolean) =>
+              (this.presentSettingMenuShow = show)}
+            .setFrameMenuShow=${(show: boolean) =>
+              (this.presentFrameMenuShow = show)}
+          ></presentation-toolbar>
+          ${type === 'frameNavigator' ? nothing : this._DefaultContent}
```

Comment:
Probably not needed `visible`. `nothing` can be replaced with `presentation-toolbar`.

Reply:
Yes, but there's a lot of initialization logic in `present-toolbar`, so this instance is preserved here and won't be destroyed in `edgeless-toolbar`.

Reply:
How about [cache](https://lit.dev/docs/templates/directives/#cache)?